eng/test-job.yml

```diff
@@ -323,6 +323,37 @@ jobs:
           - forcerelocs
           - gcstress15
 
+    # Zip tests
+    - task: ArchiveFiles@2
+      displayName: Zip Tests
+      inputs:
+        rootFolderOrFile: $(Build.SourcesDirectory)/bin/tests/$(osGroup).$(archType).$(buildConfigUpper)
+
+        ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
+          archiveType: tar
+          tarCompression: gz
+          archiveFile: $(Build.SourcesDirectory)/bin/tests/$(osGroup).$(archType).$(buildConfigUpper).tar.gz
+        ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+          archiveType: zip
+          archiveFile: $(Build.SourcesDirectory)/bin/tests/$(osGroup).$(archType).$(buildConfigUpper).zip
+        includeRootFolder: true 
+
+    - task: PublishBuildArtifacts@1
+      displayName: Publish Test Build
+      inputs:
+        ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
+          pathtoPublish: $(Build.SourcesDirectory)/bin/tests/$(osGroup).$(archType).$(buildConfigUpper).tar.gz
+        ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+          pathtoPublish: $(Build.SourcesDirectory)/bin/tests/$(osGroup).$(archType).$(buildConfigUpper).zip
+
+        ${{ if and(eq(parameters.corefxTests, true), eq(parameters.readyToRun, true)) }}:
+          artifactName: ${{ format('Tests_{0}_{1}_{2}_{3}_{4}_Logs', '_r2r_corefx', parameters.osIdentifier, parameters.archType, parameters.buildConfig, parameters.testGroup) }}
```

Comment:
I though the test publishing is going to be used for coreclr tests only? Therefore I don't think we need `_corefx` suffix.

I would also suggest to add  `publishTestArtifacts` template parameter  to test-job.yml with default value of false and set it to true only in ci.yml. I don't think we want to publish tests from the all of our pipelines.

Reply:
Is there a space concern? If not I think I would err on the side of publishing everywhere for consistency. Also publishing pri1 tests would be useful.

Reply:
Yes, I would prefer no to waste space by publishing for jitstress,jitstressregs pipelines that are going to produce exactly the same test binaries as coreclr-outerloop.

ci.yml does Pri1 testing.